app/src/main/java/com/macbitsgoa/ard/activities/helpers/AuthHelper.java

```diff
@@ -0,0 +1,175 @@
+ */
+@SuppressWarnings("WeakerAccess")
+public class AuthHelper
+        implements GoogleApiClient.OnConnectionFailedListener {
+
+    /**
+     * Tag for this class.
+     */
+    private static final String TAG = AHC.TAG + ".AuthHelper";
+
+    /**
+     * Do when signed in to firebase with credential provided by google.
+     */
+    public OnCompleteListener<AuthResult> googleAuthCompleteListener = new OnCompleteListener<AuthResult>() {
+        @Override
+        public void onComplete(@NonNull final Task<AuthResult> task) {
+            if (!task.isSuccessful()) {
+                Log.e(TAG, "authWithGoogle-onComplete: task unsuccessful", task.getException());
+                hostActivity.handleGoogleSignInFailure();
+                return;
+            }
+            updateSelfInfo();
+            hostActivity.finish();
+        }
+    };
+
+    /**
+     * GoogleApiClient object.
+     */
+    private final GoogleApiClient mGoogleApiClient;
+
+    /**
+     * Calling Activity.
+     */
+    private final AuthActivity hostActivity;
+
+    /**
+     * Constructor.
+     * @param hostActivity calling activity.
+     */
+    public AuthHelper(final AuthActivity hostActivity) {
+        this.hostActivity = hostActivity;
+        final GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken(hostActivity.getString(R.string.firebase_web_client_id))
+                .requestEmail()
+                .requestProfile()
+                .build();
+        this.mGoogleApiClient = new GoogleApiClient.Builder(hostActivity)
+                .enableAutoManage(hostActivity, this)
+                .addApi(Auth.GOOGLE_SIGN_IN_API, gso)
+                .build();
+    }
+
+    /**
+     * Constructor.
+     * @param hostActivity calling activity.
+     * @param googleApiClient {@link GoogleApiClient}.
+     */
+    public AuthHelper(final AuthActivity hostActivity, final GoogleApiClient googleApiClient) {
+        this.hostActivity = hostActivity;
+        mGoogleApiClient = googleApiClient;
+    }
+
+    /**
+     * Get FirebaseAuth instance.
+     * @return {@link FirebaseAuth} instance.
+     */
+    public FirebaseAuth getFirebaseAuth() {
+        return FirebaseAuth.getInstance();
+    }
+
+    /**
+     * Send relevant info of current user to firebase.
+     */
+    public void updateSelfInfo() {
+        final FirebaseUser user = getFirebaseAuth().getCurrentUser();
+        if (user == null) {
+            return;
+        }
+        final String uid = user.getUid();
+        final String name = user.getDisplayName();
+        final String email = user.getEmail();
+        String photoUrl = null;
+        final Uri uri = user.getPhotoUrl();
+        if (uri != null) {
+            photoUrl = uri.toString();
+        }
+        final DatabaseReference userDb = hostActivity.getRootReference().child(AHC.FDR_USERS).child(uid);
+        if (name != null) {
```

Comment:
Just the particular fields. This is done for cases when user changes signing method and that method lack some data (for ex. photoUrl). In that case, we'll not lose old data and replace it by `null` value.